projects/hello_world/src/hello_world.c

```diff
@@ -0,0 +1,33 @@
+#include <stdint.h>
+
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+#include "wait.h"
+#define INTERVAL 0.5
+
+typedef struct Counters {
+  uint8_t counter_a;
+  uint8_t counter_b;
+} Counters;
+void prv_callback(SoftTimerId id, void *context) {
```

Comment:
Nitpick: there should be a blank line between top-level elements like structs and callbacks.